chainer/functions/connection/convolution_2d.py

```diff
@@ -218,8 +227,12 @@ def backward_gpu(self, inputs, grad_outputs):
             W_mat = W.reshape(out_c, -1)
             gcol = cuda.cupy.empty_like(self.col)
             gcol_mats = gcol.reshape(n, c * kh * kw, out_h * out_w)
+
             for i in moves.range(n):
-                cuda.cupy.dot(W_mat.T, gy_mats[i], gcol_mats[i])
+                gcol_mats[i] = cuda.cupy.dot(W_mat.T, gy_mats[i])
+
+            gcol = cuda.cupy.tensordot(W, gy, (0, 1)).astype(x.dtype)
+            gcol = cuda.cupy.rollaxis(gcol, 3)
```

Comment:
Could you explain why we need additional mathematical operations, although changes so far are related to dtypes?